src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -74,11 +77,11 @@ public static void ToSerializeCall(this CodeWriter writer, JsonSerialization ser
                         case SchemaTypeReference schemaTypeReference:
                             switch (typeFactory.ResolveReference(schemaTypeReference))
                             {
-                                case ClientObject _:
+                                case ObjectType _:
```

Comment:
I don't like that we use the same suffix for type references (ISchemaType) and client models (ObjectType )